ambry-clustermap/src/test/java/com.github.ambry.clustermap/TestUtils.java

```diff
@@ -114,24 +114,44 @@ public static JSONObject getJsonDataNode(String hostname, int port, int sslPort,
     return jsonObject;
   }
 
-  // Increments basePort for each DataNode to ensure unique DataNode given same hostname.
-  public static JSONArray getJsonArrayDataNodes(int dataNodeCount, String hostname, int basePort,
+  /**
+   * Generates an array of JSON data node objects.
+   * Increments basePort and sslPort for each node to ensure unique DataNode given same hostname.
+   *
+   * @param dataNodeCount how many data nodes to generate
+   * @param hostname the hostname for each node in the array
+   * @param basePort the starting standard port number for nodes generated
+   * @param sslPort the starting SSL port number for nodes generated
+   * @param hardwareState a {@link HardwareLayout} value for each node
+   * @param disks a {@link JSONArray} of disks for each node
+   * @return a {@link JSONArray} of nodes
+   */
+  static JSONArray getJsonArrayDataNodes(int dataNodeCount, String hostname, int basePort, int sslPort,
       HardwareState hardwareState, JSONArray disks) throws JSONException {
     JSONArray jsonArray = new JSONArray();
     for (int i = 0; i < dataNodeCount; ++i) {
-      jsonArray.put(getJsonDataNode(hostname, basePort + i, hardwareState, disks));
+      jsonArray.put(getJsonDataNode(hostname, basePort + i, sslPort + i, hardwareState, disks));
     }
     return jsonArray;
   }
 
-  // Increments basePort and sslPort for each DataNode to ensure unique DataNode given same hostname.
-  public static JSONArray getJsonArrayDataNodes(int dataNodeCount, String hostname, int basePort, int sslPort,
-      HardwareState hardwareState, JSONArray disks) throws JSONException {
-    JSONArray jsonArray = new JSONArray();
-    for (int i = 0; i < dataNodeCount; ++i) {
-      jsonArray.put(getJsonDataNode(hostname, basePort + i, sslPort + i, hardwareState, disks));
+  /**
+   * Updates the given array of JSON data node objects by adding new entries to reach the total datanode count.
+   * Increments basePort and sslPort for each node to ensure unique DataNode given same hostname.
+   *
+   * @param dataNodeJsonArray the datanode JSONArray to update.
+   * @param dataNodeCount how many total datanode entries the JSONArray should have.
+   * @param hostname the hostname for each node in the array
+   * @param basePort the starting standard port number for nodes generated
+   * @param sslPort the starting SSL port number for nodes generated
+   * @param hardwareState a {@link HardwareLayout} value for each node
+   * @param disks a {@link JSONArray} of disks for each node
```

Comment:
Yes, that is how `TestUtils` is written. I don't think that matters for the purposes of testing this tool.